lib/generators/accessly/install/templates/db/migrate/create_permitted_action_on_objects.rb

```diff
@@ -0,0 +1,14 @@
+class CreatePermittedActionOnObjects < ActiveRecord::Migration<%= migration_version %>
+  def change
+    create_table :accessly_permitted_action_on_objects, force: true, id: :uuid do |t|
+      t.column :segment_id, :integer, default: -1
+      t.column :action, :integer, null: false
+      t.column :actor_id, :integer, null: false
+      t.column :actor_type, :string, null: false
+      t.column :object_type, :string, null: false
+      t.column :object_id, :integer, null: false
+    end
+
+    add_index(:accessly_permitted_action_on_objects, [:segment_id, :action, :actor_id, :actor_type, :object_type, :object_id], unique: true, name: "acessly_paoo_uniq_table_idx")
```

Comment:
### Get list of permissions an actor has on a particular lesson

#### Why
When rendering a page, we might want to conditionally render parts based on the different permissions the user has on the lesson. For example, we might render the "Edit" button only if the user has the `:edit` permission, the "Assign" button only if the user has the `:assign` permission, and the Grading Information only if the user has the `:grading` permission.

#### SQL
```sql
SELECT action
FROM accessly_permitted_action_on_objects
WHERE segment_id = $1 AND actor_type = $2 AND actor_id = $3 AND object_type = $4 AND object_id = $5
```

#### Index that might help
```rb
[:segment_id, :actor_type, :actor_id, :object_type, :object_id]
```

Reply:
@ehourigan I think we should include this index. I believe it will be a common case.

Reply:
we will move action to the end of initial index to cover this usecase